File: 2015/02/Part2.py
def ribbon(s: str) -> int:
    l,w,h = s.split("x")
    l=int(l)
    w=int(w)
    h=int(h)
    return 2*min(l+w, w+h, h+l) + l*w*h

with open("input.txt") as f:
    print(sum(ribbon(line) for line in f))

File: 2023/06/Part2.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.FileReader;

class Part2 {
  private static long processLine(String line) {
    return Long.valueOf(line.split(":")[1].replace(" ", ""));
  }

  private static void calculate(String filename) throws IOException {
    long time;
    long distance;
    try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
      time = processLine(reader.readLine());
      distance = processLine(reader.readLine());
    }
    System.out.println(Part1.numWaysToWin(time, distance));
  }

  public static void main(String[] args) throws IOException {
    if (args.length > 0) {
      calculate(args[0]);
    } else {
      calculate("input.txt");
    }
  }
}
